File: Solutions/Kamikoto66/GameShop3/Shop.cpp
#include <iostream>
#include "Shop.h"
#include "Item.h"


Shop::Shop()
{
}

Shop::Shop(std::string name, std::list<Item*> items)
	:_name(name), _items(items)
{
}


Shop::~Shop()
{
}

void Shop::ShowItemList()
{
	std::cout << _name .c_str()<< std::endl;
	std::cout << "- Item List -" << std::endl;

	for each (auto var in _items)
	{
		var->Describe();
	}
}

File: Solutions/DSMdongly/game-shop/part-3/src/game-shop/shop/shop.go
package shop

type Item interface {
	Describe()
}

type Shop struct {
	Name  string
	Items []Item
}

func NewShop(nme string, ites []Item) *Shop {
	return &Shop{
		Name:  nme,
		Items: ites,
	}
}

func (shp Shop) ShowItemList() {
	for _, ite := range shp.Items {
		ite.Describe()
	}
}
